src/watch/watch.ts

```diff
@@ -174,7 +176,7 @@ export class Task {
 				return;
 			}
 			this.updateWatchedFiles(result);
-			await Promise.all(this.outputs.map(output => result.write(output)));
+			this.skipWrite || (await Promise.all(this.outputs.map(output => result.write(output))));
```

Comment:
This is where the new option is applied.

(Not a fan of how the linter formatted it)